src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_params.py

```diff
@@ -16,12 +16,11 @@ def get_device_id_completion_list(prefix, action, parsed_args, **kwargs):#pylint
     client = iot_hub_service_factory(kwargs)
     return [d.device_id for d in iot_device_list(client, parsed_args.hub_name, top=100)] if parsed_args.hub_name else []
 
-register_cli_argument('iot hub', 'hub_name', options_list=('--name', '-n'),
-                      completer=get_resource_name_completion_list('Microsoft.Devices/IotHubs'),
-                      help='The IoT Hub name.')
-register_cli_argument('iot device', 'hub_name', options_list=('--hub-name', '--hub'),
-                      completer=get_resource_name_completion_list('Microsoft.Devices/IotHubs'),
-                      help='The IoT Hub name.')
+hub_name_type = CliArgumentType(completer=get_resource_name_completion_list('Microsoft.Devices/IotHubs'),
+                                help='IoT Hub name.')
+
+register_cli_argument('iot hub', 'hub_name', hub_name_type, options_list=('--name', '-n'))
+register_cli_argument('iot device', 'hub_name', hub_name_type, options_list=('--hub-name', '--hub'))
```

Comment:
This should be `options_list=('--hub-name',)` as we do not have multiple -- options for an argument anywhere else in the CLI. All arguments have a single long option and (optionally) a short option.

Based on existing CLI conventions, `--hub` would only be appropriate if you could pass in the name or ID of a hub. If you cannot use the ID (which it looks like you can't) then `--hub-name` should be the long option.